UsbAccess/UsbAccess.cpp

```diff
@@ -508,6 +563,172 @@ namespace Plugin {
+
+        string PartitionCount = buffer;
+
+        int ret = pclose(fp);
+        if (ret == -1) 
+        {
+            result = false;
+        }
+        response["PartitionCount"] = PartitionCount;
+        returnResponse(result);
+    }
+
+
+
+    uint32_t UsbAccess::getPartitionInfoWrapper(const JsonObject &parameters, JsonObject &response)
+    {
+        LOGINFOMETHOD();
+        bool result = true;
+        char buffer[32] = "";
+        unsigned long long int blocksize = 0;
+        JsonArray arr;
+        JsonObject ent;
+        string pathParam;
+        if (parameters.HasLabel("path"))
+            pathParam = parameters["path"].String();
+
+        struct statvfs fs;
+        if (statvfs(pathParam.c_str(), &fs) == -1) {
+            result = false;
+        }
+
+        string command_str1 = "df -T  ";
+        string command_str2 = " | tail -n +2 | awk '{print $2}'";
+        string command = command_str1 + pathParam + command_str2;
+        FILE* fp = popen(command.c_str(), "r");
+        if (fp == NULL) 
+        {
+            result = false;
+        }   
+        
+        while (fgets(buffer, 32, fp) != NULL);
+
+        string File_system = buffer;
+
+        int ret = pclose(fp);
+        if (ret == -1) 
+        {
+            result = false;
+        }
+
+        if (!result)
+        {
+            ent["error"] = "u disk device not found";
+            LOGINFO("u disk device not found\n");
+        }
+        else 
+        {
+            blocksize = fs.f_frsize /1024;
+            ent["path"] = pathParam;
+            ent["FileSystem"] = File_system;
+            ent["size(MB)"] = blocksize;
+            ent["TotalBlocks"] = fs.f_blocks;
+            ent["AvailableBlocks"] =  fs.f_bavail;
+            ent["TotalSpace(MB)"] = (fs.f_blocks * blocksize) / 1024;
+            ent["UsedSpace(MB)"] =  ((fs.f_blocks - fs.f_bavail) * blocksize) / 1024;
+            ent["AvailableSpace(MB)"] = (fs.f_bavail * blocksize) / 1024;
+        }
+
+        arr.Add(ent);
+        response["contents"] = arr;
+        returnResponse(result);    
+
+    }
+
+    uint32_t UsbAccess::getMountedInfoWrapper(const JsonObject &parameters, JsonObject &response)
+    {
+        LOGINFOMETHOD();
+        bool result = true;
+        char buffer[32] = "";
+        std::list<string> paths;
+        result = getMounted(paths);
+        JsonArray arr;
+        JsonObject ent;
+        LOGINFO("paths count = %d\n", paths.size());
+
+
+        if(paths.size() != 0)
+        {
+            for_each(paths.begin(), paths.end(), [&](const string& it)
+            {
+                string name = "";
+                string command = "findmnt -n -o source ";
+                command = command + it;
+                FILE* fp = popen(command.c_str(), "r");
+                if (fp == NULL) 
+                {
+                    result = false;
+                }   
+                
+                while (fgets(buffer, 32, fp) != NULL);
```

Comment:
Same as above comment. fp is used in logic event after check with NULL. This could cause crash.

Reply:
Done